lxd/daemon.go

```diff
@@ -2435,6 +2435,55 @@ func (d *Daemon) heartbeatHandler(w http.ResponseWriter, r *http.Request, isLead
 
 		logger.Info("Partial heartbeat received", logger.Ctx{"local": localClusterAddress})
 	}
+
+	// Refresh cluster member resource info cache.
```

Comment:
This function `heartbeatHandler` is already quite long and intricate so am not too keen on making it more so.
Please can we split this logic out into a separate function we can call from here instead. 
This is something the TIOBE TICS scoring system is also looking for (less complex functions).